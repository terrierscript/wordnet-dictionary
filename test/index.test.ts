import { test, expect } from "vitest"
// const assert = require("assert")
import {
  getLexicalEntry,
  getSynset,
  getSynsetIndex,
  getRandomWord,
  getFileIndexes,
  getFileWords
} from "../src/index"

test("getLexicalEntry", () => {
  const lex = getLexicalEntry("oewn-lie-v")
  expect(lex.sense).toEqual([
    "oewn-lie__2.42.00..",
    "oewn-lie__2.35.00..",
    "oewn-lie__2.42.01..",
    "oewn-lie__2.42.03..",
    "oewn-lie__2.32.00..",
    "oewn-lie__2.42.02..",
    "oewn-lie__2.38.00..",
  ])
})
test("synset Definition", () => {
  const synset = getSynset("oewn-02696550-v")
  // console.log(
  expect(synset.definition).toEqual([
    "be located or situated somewhere; occupy a certain position",
  ])
})

test("synsetRelation", () => {
  const synset = getSynset("oewn-02696550-v")
  // )
  // console.log(
  expect(synset.synsetRelation.map((r) => r.target)).toEqual([
    "oewn-02737265-v",
    "oewn-02661230-v",
    "oewn-02697196-v",
    "oewn-02697331-v",
    "oewn-02697564-v",
    "oewn-02697732-v",
    "oewn-02697931-v",
    "oewn-02698039-v",
    "oewn-02698177-v",
    "oewn-02698400-v",
    "oewn-02698528-v",
    "oewn-02698930-v",
    "oewn-02699010-v",
    "oewn-02699161-v",
    "oewn-02699807-v",
    "oewn-02699948-v",
    "oewn-02700129-v",
    "oewn-02701971-v",
    "oewn-02709399-v",
    "oewn-02719059-v",
    "oewn-02727841-v",
  ])
})
test("synset members", () => {
  const synset2 = getSynsetIndex("oewn-02709399-v")
  // console.log(
  //   assert.deepStrictEqual(
  expect(synset2.lexicalEntry).toEqual(["oewn-line-v", "oewn-run_along-v"])
  // ?? "âœ… synset members"
  // // )
  // // searchSynset()
})
test("random", () => {
  for (let i = 0;i < 100;i++) {
    const random = getRandomWord()
    expect(typeof random).toBe("string")
  }
})

test("getFileIndexes", () => {
  expect(getFileIndexes()).toMatchSnapshot()
})

test("getFileWords", () => {
  expect(getFileWords("00")).toMatchSnapshot()
})

