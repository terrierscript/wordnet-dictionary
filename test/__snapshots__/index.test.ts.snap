// Vitest Snapshot v1

exports[`getFileIndexes 1`] = `
[
  "00",
  "01",
  "02",
  "03",
  "04",
  "05",
  "06",
  "07",
  "08",
  "09",
  "0a",
  "0b",
  "0c",
  "0d",
  "0e",
  "0f",
  "10",
  "11",
  "12",
  "13",
  "14",
  "15",
  "16",
  "17",
  "18",
  "19",
  "1a",
  "1b",
  "1c",
  "1d",
  "1e",
  "1f",
  "20",
  "21",
  "22",
  "23",
  "24",
  "25",
  "26",
  "27",
  "28",
  "29",
  "2a",
  "2b",
  "2c",
  "2d",
  "2e",
  "2f",
  "30",
  "31",
  "32",
  "33",
  "34",
  "35",
  "36",
  "37",
  "38",
  "39",
  "3a",
  "3b",
  "3c",
  "3d",
  "3e",
  "3f",
  "40",
  "41",
  "42",
  "43",
  "44",
  "45",
  "46",
  "47",
  "48",
  "49",
  "4a",
  "4b",
  "4c",
  "4d",
  "4e",
  "4f",
  "50",
  "51",
  "52",
  "53",
  "54",
  "55",
  "56",
  "57",
  "58",
  "59",
  "5a",
  "5b",
  "5c",
  "5d",
  "5e",
  "5f",
  "60",
  "61",
  "62",
  "63",
  "64",
  "65",
  "66",
  "67",
  "68",
  "69",
  "6a",
  "6b",
  "6c",
  "6d",
  "6e",
  "6f",
  "70",
  "71",
  "72",
  "73",
  "74",
  "75",
  "76",
  "77",
  "78",
  "79",
  "7a",
  "7b",
  "7c",
  "7d",
  "7e",
  "7f",
  "80",
  "81",
  "82",
  "83",
  "84",
  "85",
  "86",
  "87",
  "88",
  "89",
  "8a",
  "8b",
  "8c",
  "8d",
  "8e",
  "8f",
  "90",
  "91",
  "92",
  "93",
  "94",
  "95",
  "96",
  "97",
  "98",
  "99",
  "9a",
  "9b",
  "9c",
  "9d",
  "9e",
  "9f",
  "a0",
  "a1",
  "a2",
  "a3",
  "a4",
  "a5",
  "a6",
  "a7",
  "a8",
  "a9",
  "aa",
  "ab",
  "ac",
  "ad",
  "ae",
  "af",
  "b0",
  "b1",
  "b2",
  "b3",
  "b4",
  "b5",
  "b6",
  "b7",
  "b8",
  "b9",
  "ba",
  "bb",
  "bc",
  "bd",
  "be",
  "bf",
  "c0",
  "c1",
  "c2",
  "c3",
  "c4",
  "c5",
  "c6",
  "c7",
  "c8",
  "c9",
  "ca",
  "cb",
  "cc",
  "cd",
  "ce",
  "cf",
  "d0",
  "d1",
  "d2",
  "d3",
  "d4",
  "d5",
  "d6",
  "d7",
  "d8",
  "d9",
  "da",
  "db",
  "dc",
  "dd",
  "de",
  "df",
  "e0",
  "e1",
  "e2",
  "e3",
  "e4",
  "e5",
  "e6",
  "e7",
  "e8",
  "e9",
  "ea",
  "eb",
  "ec",
  "ed",
  "ee",
  "ef",
  "f0",
  "f1",
  "f2",
  "f3",
  "f4",
  "f5",
  "f6",
  "f7",
  "f8",
  "f9",
  "fa",
  "fb",
  "fc",
  "fd",
  "fe",
  "ff",
]
`;

exports[`getFileWords 1`] = `
[
  "A-line",
  "Abnaki language",
  "Air Force Intelligence Surveillance and Reconnaissance",
  "Akeridae",
  "Alfred Lunt",
  "Alosa pseudoharengus",
  "American cress",
  "Anshar",
  "Araliaceae",
  "Arctic ground squirrel",
  "Arctictis",
  "Ascophyllum",
  "B-flat clarinet",
  "BNS",
  "Babyrousa Babyrussa",
  "Basidiomycota",
  "Batis maritima",
  "Baton Rouge Bridge",
  "Boston baked beans",
  "Brachychiton populneus",
  "Brahui language",
  "Bryonia alba",
  "Caenolestes",
  "Calamintha grandiflora",
  "Calandrinia ciliata",
  "Callithricidae",
  "Camelus",
  "Capra",
  "Carolina moonseed",
  "Carroll",
  "Cassia marilandica",
  "Cassius Marcellus Clay",
  "Ceratopetalum",
  "Chadian",
  "Chagatai language",
  "Charles Farrar Browne",
  "Cimicifuga americana",
  "Citlaltepetl",
  "Claviceps",
  "Clitocybe robusta",
  "Colorado beetle",
  "Continental Divide of the Americas",
  "Copernican system",
  "Cork",
  "Cupressus lusitanica",
  "Dante Gabriel Rossetti",
  "Daphne",
  "Defoe",
  "Edwin DuBois Hayward",
  "Eleaticism",
  "English hawthorn",
  "Euplectella",
  "Federated States of Micronesia",
  "Georg Ohm",
  "German silver",
  "Gianbattista Bodoni",
  "God's acre",
  "Gordon setter",
  "Gossypium herbaceum",
  "Hemerocallis lilio-asphodelus",
  "Hephaistos",
  "Hermosillo",
  "Hieracium pilocella",
  "Holocephali",
  "Icelandic-speaking",
  "Ilya Ilich Metchnikov",
  "Isoptera",
  "Jacques Charles",
  "Japanese persimmon",
  "Joachim",
  "John Herschel",
  "Julius Winfield Erving",
  "Karl Adolph Gjellerup",
  "Korea Strait",
  "Leonard Bloomfield",
  "Lords Temporal",
  "Louis XV of France",
  "Malti language",
  "Mandarin dialect",
  "Medan",
  "Micrococcus",
  "Monroe",
  "Montreal",
  "Mutsuhito",
  "Naias",
  "Neomys fodiens",
  "Orpheus",
  "Oryza sativa",
  "PDA",
  "PFLOPS",
  "Palestine",
  "Parker House roll",
  "Parnahiba",
  "Pennsylvania Dutch",
  "People's Mujahedin of Iran",
  "Peripatidae",
  "Philippopolis",
  "Picea mariana",
  "Picris",
  "Platalea",
  "Polystichum lonchitis",
  "Prosopis glandulosa",
  "Rangifer arcticus",
  "Raoulia",
  "Red Guard",
  "Rhododendron viscosum",
  "Riksmaal",
  "River Styx",
  "Rosellinia",
  "Rossetti",
  "Rotary International",
  "Samuel Wilder",
  "Satureja grandiflora",
  "Scaphiopus hammondii",
  "Scotch egg",
  "Selective Service System",
  "Shabuoth",
  "Shawnee",
  "Shoshonian language",
  "Sir Henry Morton Stanley",
  "Sir John Vanbrigh",
  "Sir William Turner Walton",
  "Slovenia",
  "Snellen chart",
  "Sulidae",
  "Supreme Allied Commander Atlantic",
  "Suricata suricatta",
  "Sylvilagus aquaticus",
  "Taccaceae",
  "Talinum brevifolium",
  "Tamiasciurus",
  "Teasdale",
  "Treaty of Versailles",
  "Upupidae",
  "Velveeta",
  "Virgo the Virgin",
  "Volkhov",
  "Wandala",
  "Weimar",
  "Welwitschia mirabilis",
  "Zarqa",
  "academic costume",
  "accretion",
  "acned",
  "adenocarcinomata",
  "advertizing",
  "agricultural agent",
  "air-breathing",
  "algarroba",
  "aliterate",
  "alphanumerics",
  "amblyopic",
  "anthracite",
  "aquiline",
  "archosaur",
  "ariette",
  "armor-bearer",
  "atomic number 30",
  "atomic number 45",
  "axiological",
  "backtalk",
  "basal body temperature",
  "bay leaf",
  "beaked salmon",
  "beam-ends",
  "bed",
  "begrimed",
  "binding",
  "biological clock",
  "birdbrain",
  "blackface",
  "bloodcurdling",
  "bortsch",
  "breakwater",
  "bring round",
  "broken-down",
  "brougham",
  "bullfinch",
  "bye-bye",
  "cabinet minister",
  "caddis fly",
  "calcium hydride",
  "caldera",
  "calendar method of birth control",
  "canella bark",
  "canulization",
  "capitulation",
  "cardiopathy",
  "caressing",
  "carrot",
  "caseous",
  "chalazae",
  "change magnitude",
  "charge account",
  "charity toss",
  "chatroom",
  "checking program",
  "chicanery",
  "chopping board",
  "chrome-nickel steel",
  "chuck out",
  "cigarette smoker",
  "class Euglenophyceae",
  "clean room",
  "clear sailing",
  "clenched fist",
  "cloying",
  "clung",
  "cogitative",
  "coiffeuse",
  "commutative group",
  "commuter traffic",
  "comp",
  "concave lens",
  "concrete",
  "conscienceless",
  "corespondent",
  "corner pocket",
  "cornmeal mush",
  "cornstalk",
  "crank handle",
  "cresol",
  "damask violet",
  "danse macabre",
  "dark red",
  "dawn",
  "deep kiss",
  "defervescence",
  "demolishing",
  "departer",
  "despite",
  "deviationism",
  "disobedient",
  "disport",
  "dissever",
  "dissimulator",
  "divergency",
  "diving suit",
  "dolman sleeve",
  "domestic dog",
  "douglas spruce",
  "driven well",
  "drubbing",
  "easiest",
  "eat up",
  "ecce homo",
  "eclectic",
  "efferent fiber",
  "electric organ",
  "electrified",
  "elongate",
  "enfranchise",
  "ensiform leaf",
  "entrance examination",
  "eptatretus",
  "equivocation",
  "estimation theory",
  "eternity",
  "eugenics",
  "euglenoid",
  "evaporated",
  "expansivity",
  "facade",
  "family Bradypodidae",
  "family Hydrobatidae",
  "family Poaceae",
  "feasting",
  "fifty-fifth",
  "finback whale",
  "fish loaf",
  "flatmate",
  "foreperson",
  "forrard",
  "four o'clock",
  "frap",
  "frontal bone",
  "galosh",
  "gasoline pump",
  "genital phase",
  "genus Apis",
  "genus Artamus",
  "genus Crassostrea",
  "genus Crocodilus",
  "genus Haemanthus",
  "genus Koellia",
  "genus Lagerstroemia",
  "genus Maianthemum",
  "genus Parathelypteris",
  "genus Podiceps",
  "genus Scomberesox",
  "genus Toxicodendron",
  "geographically",
  "geothermally",
  "germaneness",
  "gizmo",
  "glib-tongued",
  "gluon",
  "gnostic",
  "golden section",
  "golf caddie",
  "grass bacillus",
  "gray-haired",
  "gray-pink",
  "guidebook",
  "haj",
  "hammed",
  "hand cheese",
  "handheld drill",
  "handlock",
  "handmaiden",
  "hanging chad",
  "happen",
  "hard-and-fast",
  "hatband",
  "hatchway",
  "headfirst",
  "hearing-impaired",
  "heart attack",
  "hectometer",
  "hedonistic",
  "hematocolpometra",
  "hobbyhorse",
  "hole-in-the-wall",
  "home theatre",
  "homogenized",
  "honeymooner",
  "honorary",
  "horse-and-buggy",
  "hospitaller",
  "house organ",
  "hybridizing",
  "ictodosaur",
  "impressment",
  "imprudently",
  "inclined plane",
  "incurability",
  "indene",
  "indicatory",
  "infectious disease",
  "infer",
  "inference",
  "infinitely",
  "insecticide",
  "intersection point",
  "inventory-clearance sale",
  "jennet",
  "jointer plane",
  "joviality",
  "jumpsuit",
  "juristic",
  "kenning",
  "king vulture",
  "kirtle",
  "knee joint",
  "ladies' tresses",
  "larder",
  "leave alone",
  "length of service",
  "let in",
  "light-emitting diode",
  "linguistic rule",
  "literally",
  "lithomantic",
  "logistic support",
  "lousiest",
  "lxv",
  "lyrical",
  "maimed",
  "maladroitness",
  "manakin",
  "matchlessness",
  "maximally",
  "memory apparatus",
  "mensch",
  "mental strain",
  "metamorphopsia",
  "metamorphose",
  "miner's cat",
  "misadventure",
  "mishap",
  "monitrice",
  "mope around",
  "mosquito craft",
  "mountain anemone",
  "mountain beaver",
  "mucoidal",
  "mulattoes",
  "nerve fiber",
  "nervus ischiadicus",
  "neutralization",
  "nihility",
  "nociceptive",
  "nomenclature",
  "nonmechanistic",
  "norethandrolone",
  "nose dive",
  "nutmeg family",
  "oarsmanship",
  "obituary",
  "objective case",
  "ogle",
  "olive branch",
  "ominously",
  "ommatidia",
  "once in a while",
  "one-sixtieth",
  "ooh",
  "order Cydippidea",
  "order Passeriformes",
  "output contract",
  "oval-bodied",
  "overall",
  "ovoflavin",
  "packaged goods",
  "paisley",
  "palatine",
  "palm nut",
  "paralanguage",
  "parous",
  "peaked",
  "pediculicide",
  "pep pill",
  "perkily",
  "personhood",
  "perspiration",
  "perspire",
  "philosophers' stone",
  "photograph album",
  "piaffe",
  "pigeon loft",
  "pigeon-breasted",
  "pinkness",
  "pinna",
  "piolet",
  "pistol shrimp",
  "pitch pipe",
  "pitcher's mound",
  "polar circle",
  "pole star",
  "popularization",
  "posterior serratus muscle",
  "postmaster",
  "postoperatively",
  "pout",
  "pre-valentines day",
  "presumable",
  "prey",
  "prolegomena",
  "prolixity",
  "prototypic",
  "public relations",
  "public transit",
  "purple ground cherry",
  "purplish-blue",
  "pushful",
  "quadrumvirate",
  "rearing",
  "recognized",
  "recommencement",
  "record album",
  "regain",
  "remediable",
  "remote-controlled",
  "replaceable",
  "repositing",
  "retail merchant",
  "retinoblastoma",
  "retrograde amnesia",
  "revenue tariff",
  "reverberatory furnace",
  "reversal",
  "rigmarole",
  "robust",
  "rocket salad",
  "rootlet",
  "rotational latency",
  "rotavirus",
  "rough-house",
  "rust fungus",
  "sangapenum",
  "scarabaeus",
  "scourge",
  "sea scooter",
  "self-fertilized",
  "seraphical",
  "serotonin",
  "seventh chord",
  "shakeup",
  "sheathing",
  "sheep bell",
  "shingly",
  "shiny lyonia",
  "shoot one's mouth off",
  "shoulder in",
  "shrivelling",
  "silicle",
  "sinoatrial node",
  "six-spot",
  "slate-grey",
  "slipper orchid",
  "snakier",
  "soupspoon",
  "sour-gum family",
  "source materials",
  "spinal fluid",
  "spirally",
  "splodge",
  "sports desk",
  "sprung",
  "sq ft",
  "stave off",
  "steam locomotive",
  "stirrup",
  "stock exchange",
  "storm centre",
  "straightness",
  "stroppy",
  "subtractor",
  "suit of armour",
  "supervene upon",
  "support column",
  "swamp horsetail",
  "switch-hitter",
  "tangle",
  "tartness",
  "tchotchkeleh",
  "tear into",
  "technique",
  "tediously",
  "teniae",
  "territorialize",
  "three-petalled",
  "tickle pink",
  "tidewater",
  "timberman",
  "tin foil",
  "to perfection",
  "toasting",
  "tomato",
  "tonic epilepsy",
  "tour of duty",
  "transferrable",
  "transversally",
  "tugrik",
  "twentieth century",
  "unartistic",
  "unconditioned",
  "undersexed",
  "unendowed",
  "unfeminine",
  "unheated",
  "unicameral",
  "universal suffrage",
  "university extension",
  "unleaded petrol",
  "unplug",
  "unredeemed",
  "unseen",
  "ununhexium",
  "urethrae",
  "vanish",
  "vapor density",
  "velvet-textured",
  "vena brachiocephalica",
  "vinblastine",
  "viola da gamba",
  "waldmeister",
  "wall barley",
  "water fountain",
  "waxwork",
  "whatever",
  "whelm",
  "winter break",
  "worried",
  "xanthopsia",
  "xiphoid process",
  "yellow mountain saxifrage",
  "youngest",
  "youngness",
  "ziggurat",
  "zinc ointment",
]
`;
